pkg/resource/interfaces.go

```diff
@@ -176,6 +176,18 @@ type ConnectionDetailsPublishedTimer interface {
 	GetConnectionDetailsLastPublishedTime() *metav1.Time
 }
 
+// ReconciliationObserver can track data observed by resource reconciler.
+type ReconciliationObserver interface {
+	SetObservedGeneration(generation int64)
+	GetObservedGeneration() *int64
+
+	SetObservedLabels(labels map[string]string)
+	GetObservedLabels() map[string]string
+
+	SetObservedAnnotations(annotations map[string]string)
+	GetObservedAnnotations() map[string]string
+}
```

Comment:
Do you expect that managed resources would also satisfy this interface, eventually?